python/dgl/nn/pytorch/explain/subgraphxexplainer.py

```diff
@@ -0,0 +1,566 @@
+        has a high total reward value, a high coef will cause the algorithm to explore other nodes
+        instead of continuing to visit that node.
+    num_visit: int
+        Number times this node has been visited.
+    total_reward: float
+        The total reward for all node visits.
+    immediate_reward: float
+        Immediate reward for selecting this node (property score).
+    """
+
+    def __init__(
+        self,
+        nodes,
+        high2low,
+        coef=10.0,
+        num_visit=0,
+        total_reward=0.0,
+        immediate_reward=0.0,
+    ):
+        self.nodes = nodes
+        self.high2low = high2low
+        self.coef = coef
+        self.num_visit = num_visit
+        self.total_reward = total_reward
+        self.immediate_reward = immediate_reward
+        self.children = []
+
+    def average_reward(self):
+        r"""Get the average reward for multiple visits.
+
+        Returns
+        -------
+        float
+            Return the averaged reward for multiple visits.
+        """
+        return self.total_reward / self.num_visit if self.num_visit > 0 else 0
+
+    def action_selection_criteria(self, total_visit_count):
+        r"""Get the action selection criteria of node.
+
+        Parameters
+        ----------
+        total_visit_count: float
+            The total visiting counts for all possible actions of node.
+
+        Returns
+        -------
+        float
+            Returns the action selection criteria of node.
+        """
+        return (
+            self.coef
+            * self.immediate_reward
+            * math.sqrt(total_visit_count)
+            / (1 + self.num_visit)
+        )
+
+
+class SubgraphXExplainer(nn.Module):
+    r"""SubgraphXExplainer model from `SubgraphXExplainer: On Explainability of
+    Graph Neural Networks via Subgraph Explorations <https://arxiv.org/pdf/2102.05152.pdf>`.
+
+    It identifies the most important subgraph from the original graph that plays a
+    critical role in GNN-based graph classification.
+
+    It employs Monte Carlo tree search (MCTS) in efficiently exploring different subgraphs
+    for explanation and uses Shapley values as the measure of subgraph importance.
+
+    Parameters
+    ----------
+    model: nn.Module
+        The GNN model to explain.
+
+        * The required arguments of its forward function are `DGLGraph` graph and its
+        node features.
+        * The output of its forward function is the logits for the predicted
+          graph classes.
+    num_gnn_layers: int
+        Number of layers of GNN model. Needed to calculate the l-hop neighbouring
+        nodes of graph. Computing shapley values for big and complex graphs
+        is time and resource consuming. The shapley algorithm implemented in the paper
+        efficiently approximate shapley values by finding coalition of l-hop neighboring
+        nodes of the graph, where the GNN model has l layers.
```

Comment:
```suggestion
        nodes of the graph, where the GNN model has l GNN layers.
```
A GNN can have multiple linear layers in a GNN layer.

Reply:
done.